src/test/java/me/karakelley/http/handlers/NewResourceStrategyTest.java

```diff
@@ -12,14 +13,14 @@
 
 import static org.junit.jupiter.api.Assertions.*;
 
-class NewResourceHandlerTest {
+class NewResourceStrategyTest {
```

Comment:
I think these strategies are probably implementation details. I would test all this behavior through `StaticFilesHandler`. Then you're free to refactor within that boundary. If you decide you want to pull common behavior up into that `StaticFilesHandler` or merge all these into one class you don't want to be breaking all these tests.

Reply:
This is probably what I would use that nested class syntax in JUnit 5 for (a nested context for each method). I think this syntax is controversial, but I like it...

```
public class StaticFilesHandlerTest {
  @Nested
  public class when_the_request_is_a_get {
    @Nested
    public class when_the_requested_resource_exists {
      public void sets_the_file_contents_as_the_body() {
        //...
      }

      public void sets_the_content_type_to_application_html_for_files_with_an_html_file_extension() {
        //...
      }
    }
  }
}